validations/src/validations.rs

```diff
@@ -103,28 +55,78 @@ pub fn transaction_outputs_sum(tx: &TransactionBody) -> u64 {
 /// of the transaction. The pool parameter is used to find the
 /// outputs pointed by the inputs and that contain the actual
 /// their value.
-pub fn transaction_fee(tx: &TransactionBody, utxo_diff: &UtxoDiff) -> Result<u64, failure::Error> {
-    let in_value = transaction_inputs_sum(tx, utxo_diff)?;
-    let out_value = transaction_outputs_sum(tx);
+pub fn transaction_fee(
+    tx: &Transaction,
+    utxo_diff: &UtxoDiff,
+    dr_pool: &DataRequestPool,
+) -> Result<u64, failure::Error> {
+    match tx {
+        Transaction::ValueTransfer(vt_tx) => {
+            let in_value = transaction_inputs_sum(&vt_tx.body.inputs, utxo_diff)?;
+            let out_value = transaction_outputs_sum(&vt_tx.body.outputs);
+
+            if out_value > in_value {
+                Err(TransactionError::NegativeFee)?
+            } else {
+                Ok(in_value - out_value)
+            }
+        }
+        Transaction::DataRequest(dr_tx) => {
+            let in_value = transaction_inputs_sum(&dr_tx.body.inputs, utxo_diff)?;
+            let out_value = dr_tx.body.dr_output.value;
+
+            if out_value > in_value {
+                Err(TransactionError::NegativeFee)?
+            } else {
+                Ok(in_value - out_value)
+            }
+        }
 
-    if out_value > in_value {
-        Err(TransactionError::NegativeFee)?
-    } else {
-        Ok(in_value - out_value)
+        // TODO: Miner fee is for all the commits and reveals no by each one
```

Comment:
Yes, so we can just divide the commit fee by the number of witnesses as specified in the data request